cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -812,6 +819,10 @@ func ensureNodeMatchesMetadataOrDelete(ctx *controllerContext, csr *capi.Certifi
 
 	recordMetric = csrmetrics.OutboundRPCStartRecorder("k8s.Nodes.delete")
 	err = ctx.client.CoreV1().Nodes().Delete(context.TODO(), nodeName, metav1.DeleteOptions{Preconditions: metav1.NewUIDPreconditions(string(node.UID))})
+	if err := deleteAllPodsBoundToNode(ctx, nodeName); err != nil {
```

Comment:
Let's also add a comment here to explain (similar as above):

```
GCE MIGs currently reuse the instance name on new VMs. For example, after GCE Preemptible/Spot VM is preempted, the instance started by MIG will have the same instance name. This can result in old "stale" pods still bound to the node to exist after a node is preempted.

In the case that a GCE node is preempted and new GCE instance is created with the same name, explicitly delete all bounds to the old node.
```